### ISSUE TITLE
Questions about real-world image training

### ISSUE THREAD

**Issue by FUNGERN** [hongwang01/RCDNet#2]
Questions about real-world image training
Hello, thank you very much for your contribution. I am training spa files. I downloaded real_world", "real_world_gt" and "real_world.txt" according to the link you gave you, and unzipped the compressed package into a folder, as shown in the figure Show ("real_world.txt" is provided in your code).
![1](https://user-images.githubusercontent.com/49382317/97377455-0edaa480-18fb-11eb-9029-7cbcf81e3143.png)
![image](https://user-images.githubusercontent.com/49382317/97377565-53fed680-18fb-11eb-8f12-2a55c86be2b4.png)

But when I started training, I started to report an error and a problem occurred, as shown in the figure.
![image](https://user-images.githubusercontent.com/49382317/97377550-48abab00-18fb-11eb-891c-72518a984ee8.png)

Do you know what the solution is? If you know, please let me know.

**Comment by FUNGERN** [comment]
![image](https://user-images.githubusercontent.com/49382317/97378201-c4f2be00-18fc-11eb-9e06-7d120b5326c6.png)
When I added "/" in the corresponding position of "trainer.py", a new error appeared. 
![image](https://user-images.githubusercontent.com/49382317/97378397-3e8aac00-18fd-11eb-9633-6455ec168ed7.png)

I felt that "/" should not be added, but I don't know how to solve FileNotFoundError: [Errno 2] No such file or directory:'../datareal_world.txt' problem, please tell me, thanks

**Comment by hongwang01** [comment]
Thanks for your attention on our work !

I have checked this error and fixed the bugs. You can try it again !  

If any question, feel free to concat me ! Thanks!

**Comment by FUNGERN** [comment]
> Thanks for your attention on our work !
> 
> I have checked this error and fixed the bugs. You can try it again !
> 
> If any question, feel free to concat me ! Thanks!

Thank you for your solution, but after I updated trainer.py, I encountered a new problem.When I was training for the third epoch, this situation occurred, and I have tried many times, but this situation occurred. I searched for a solution, but still did not solve the problem.Do you know this solution? thx!
![Screenshot from 2020-11-01 09-47-39](https://user-images.githubusercontent.com/49382317/97793510-eff44f00-1c27-11eb-9ad1-16159be3dac2.png)

**Comment by hongwang01** [comment]
I have not encountered such a problem. I have also trained it agrain and the training process has  successfully reached the 18th epoch. Maybe you need to check your server and the environment configuration. Actually, we have released the pretrained models for several benchmark datasets for ease of usage.

![trainspa](https://user-images.githubusercontent.com/43700818/97823463-0962cd00-1cf4-11eb-8b31-161cc718437e.png)

**Comment by hongwang01** [comment]
Now we‘re extending this work. It is expected that in the next  month we will be able to publish a simplified version for RCDNet.  Please stay tuned ！Thanks !

**Comment by FUNGERN** [comment]
I tried a lot of methods. I split all the pictures in the training set once. I found that one of the pictures in the GT could not be read out. I took real_ It was replaced by the same image in the world, but the training was interrupted at the third epoch. Just yesterday, I was real_ world.txt When there are only 1000 paths left, he can train and train to the 24th epoch, but the effect becomes very poor. I guess it may be real_ world.txt The problem.

> 现在，我们正在扩展这项工作。预期在下个月我们将能够发布RCDNet的简化版本。请继续关注！谢谢！

**Comment by FUNGERN** [comment]
> 现在，我们正在扩展这项工作。预期在下个月我们将能够发布RCDNet的简化版本。请继续关注！谢谢！

I have an idea. Can you provide the 44g t data set in your computer? I feel that there may be some problems when I download the images of the dataset from the original website. If you can, thank you very much!

**Comment by hongwang01** [comment]
The original download link: https://drive.google.com/drive/folders/1yeoqiGEK4ksq1JDYK3ImbyazHgztatEb?usp=sharing

You'd better check whether the data has been downloaded completely and whether the files have been decompressed correctly. If you still have any questions about SPA-Data, we suggest you ask the original author for help.

**Comment by FUNGERN** [comment]
> The original download link: https://drive.google.com/drive/folders/1yeoqiGEK4ksq1JDYK3ImbyazHgztatEb?usp=sharing
> 
> You'd better check whether the data has been downloaded completely and whether the files have been decompressed correctly. If you still have any questions about SPA-Data, we suggest you ask the original author for help.

`cat` real_world.zip* >  real_world.zip
`unzip` real_world.zip
 right？

**Comment by hongwang01** [comment]
Just follow the guidance in this original link above !  Any question, please ask the original author!

**Comment by hongwang01** [comment]
> I have not encountered such a problem. I have also trained it agrain and the training process has successfully reached the 18th epoch. Maybe you need to check your server and the environment configuration. Actually, we have released the pretrained models for several benchmark datasets for ease of usage.
> 
> ![trainspa](https://user-images.githubusercontent.com/43700818/97823463-0962cd00-1cf4-11eb-8b31-161cc718437e.png)

**Comment by szpxmu** [comment]
i meet the same problem.

**Comment by szpxmu** [comment]
I have solve this problem by use "img.imread" to replace "cv2.read"
And Y need add the lib "import matplotlib.image as img"
if y change the code. y will find a problem in 274-14 image name is a mistake. Y need to change the right name for image read.

**Comment by hongwang01** [comment]
> I have solve this problem by use "img.imread" to replace "cv2.read" And Y need add the lib "import matplotlib.image as img" if y change the code. y will find a problem in 274-14 image name is a mistake. Y need to change the right name for image read.

I do not think it is a matter of how the image is read.  From the bug, it should be the rainy image "O" has not been loaded correctly.

**Comment by hongwang01** [comment]
> I have solve this problem by use "img.imread" to replace "cv2.read" And Y need add the lib "import matplotlib.image as img" if y change the code. y will find a problem in 274-14 image name is a mistake. Y need to change the right name for image read.

As far as I know, the author modified the public spa-data. So it's recommended to double check whether the downloaded dataset information and the dataloader config match or not.
